src/test/java/core/basesyntax/service/impl/ReadServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.ValidationException;
+import core.basesyntax.service.ReadService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ReadServiceImplTest {
+    private final ReadService readService = new ReadServiceImpl();
+    private Path tempFile;
+    
+    @BeforeEach
+    void setUp() throws IOException {
+        tempFile = Files.createTempFile("test", ".csv");
+    }
+    
+    @AfterEach
+    void tearDown() throws IOException {
+        Storage.fruitShopData.clear();
+        Files.deleteIfExists(tempFile);
+    }
+    
+    @Test
+    void add_correctData_Ok() throws IOException {

```

Comment:
Why all methods start with add?
We're testing fileReader in this class, so readFile is much more obvious